docs/end_to_end_tutorials/dev_practices/component_wise_evaluation.md

```diff
@@ -0,0 +1,29 @@
+# Component Wise Evaluation
```

Comment:
i think we should describe which components we're evaluating:
- retrieval?
- LLM? 
- other stuff?

Reply:
ditto with above

Reply:
can we still add an intro text here framing why we should evaluate components in isolation?